lib/ansible/module_utils/common/file.py

```diff
@@ -89,7 +89,7 @@ def set_lock(self, path, tmpdir, lock_timeout=None):
 
         self.lockfd = open(lock_path, 'w')
 
-        if lock_timeout <= 0:
+        if lock_timeout == 0:
```

Comment:
Since the `lineinfile` defaults to `-1` for the `lock_timeout`, this conditional is skipped but the next matches and an exception is raised. Since `0` is less than `-1` but `-1` is "truthy", the while loop is skipped, the lockfile is immediately closed, and an exception as raised.

I believe setting the default for `lock_timeout` to 0 or possibly a small positive integer in the `argument_spec` would be the best solution. Then, this change would not be necessary.

Reply:
Another option would be to change the default of `lock_timeout` to `0` in the method signature.

Reply:
Ah, you're right!
Problem here is that in python3 using `lock_timeout <= 0` throws a TypeError exception  ('<=' not supported between instances of 'NoneType' and 'int') if the default value is left as is. It however works fine on python2, which of course was used while testing FileLock. Anyway, at the time it seemed more logical to have None as default for lock_timeout as it indicates that timeout shouldn't be used at all, I.E wait forever, this however mixes types (None and int) which is a bit inelegant and makes the ansible-validation angry. I was thinking that 0 should probably indicate that the timeout should be 0 and fail immediately if a lock is blocking.